erpnext/erpnext_integrations/taxjar_integration.py

```diff
@@ -364,28 +376,32 @@ def sanitize_error_response(response):
 
 	return response
 
-
-def get_line_items(doc):
-	if not doc.items:
-		return
-
-	discout_percent = 0.0
+def get_doc_discount(doc):
+	""""
+		It will take doc discount which is applied on Net Total/ Grand Total. It will distribute discount to line item based on qty
+	"""
+	discount = 0
+	qty = 0
+	for item in doc.items:
+		qty = qty + item.get('qty')
 
 	if doc.discount_amount:
-		discout_percent = ((doc.discount_amount / doc.total) * 100)
-
-	line_items = []
-	for item in doc.items:
-		if discout_percent:
-			item.discount_amount = ((item.price_list_rate * discout_percent) / 100)
-		line_item = {
-			'id': item.name,
-			'description': item.item_name,
-			'quantity': item.qty,
-			'unit_price': item.price_list_rate,
-			'discount': round(item.discount_amount * item.qty)
-		}
-		line_items.append(line_item)
-
-	return line_items
+		discount = doc.discount_amount/qty
```

Comment:
this would be true only if the discount is applied on Net Total. If the discount is applied on Grand Total this logic would be incorrect.

Reply:
look at my close PR https://github.com/Bloomstack/erpnext/pull/1831. Where i did deducting shipping discount from shipping parameter. but problem is taxjar try to match line item total - discount + shipping should equal to suborder Total(amount) where tax jar failed.
why, this is because lets take 
line item 1 = 100 
line item 2 = 100 but with discount 50% = 50 (unit price =100, discount = 50)
shipping amount = 100
sales tax = 7.50

if we apply a discount on grand total (let take 20%)
it will distribute discount on line item = 150

use case 1: all distribute
grandTotal before discount = 257.5*20%=  257.5 -51.5 = 206

use case 2: deduct shipping first and distribute
51.5 - 100*20% = 31.5 need to distribute and from 100 shipping remove 20 


so 31.5/2 = 15.75 need to distribute to line item = 150+31.5= 181.5 this is line item total + shipping 80 = 261.5 and need to match with 200+80=280